lua/lazyvim/plugins/extras/lang/rust.lua

```diff
@@ -42,7 +42,7 @@ return {
 
   {
     "mrcjkb/rustaceanvim",
-    version = "^4", -- Recommended
+    version = vim.fn.has("nvim-0.10.0") == 0 and "^4" or false,
```

Comment:
Maybe is better to use `vim.version()`.

```suggestion
    version = (vim.version().major == 0 and vim.version().minor == 9) and "^4" or false,
```

Reply:
Agree.

Reply:
`vim.fn.has` is more consise and used throughout many plugins to check for features. It's more of a convention to use it.

Reply:
I see that this is a subjective thing, so I'll use the convention @folke tells me to. :)

Reply:
https://github.com/search?q=repo%3ALazyVim%2FLazyVim%20vim.version&type=code

https://github.com/search?q=repo%3ALazyVim%2FLazyVim%20vim.fn.has&type=code

The only usage of `vim.version` is to construct `0_9` as a string to be used in the compat import for Neovim versions 0.9+

Reply:
If the user has Neovim version `0.9.5`, it means he will have the latest version of Rustacean Vim (not `^4`).

Reply:
`vim.fn.has("nvim-0.10.0")` returns 0 unless the user is on 0.10 or above.

Reply:
Where do you base this off? I just tested Neovim 0.9.5 and `vim.fn.has("nvim-0.10")` returns `0`.

Reply:
Exactly, that's what I said. :) Then it will use 4.

Reply:
I also tested this PR locally with Neovim 0.9.5 and `rustaceanvim` is 4.26.1

![2024-09-27_14-30](https://github.com/user-attachments/assets/32428db5-1893-4f0d-8cb2-394ea5c24c85)

Reply:
@edevil Sorry my previous comment wasn't towards you, rather towards @IvanProg00. I'm just mentioning it, so that there will be no confusion.